### PR TITLE
feat: Redux leaderboard & battletv page

### PR DESCRIPTION
**Pull request by jaiakash**
- [x] Fetch from server to display on leaderboard page [Link](https://rinish-api-testbed.stoplight.io/docs/codecharacter/b3A6MTQwMzYxOTk-get-leaderboard)
- [x] Fetch from server to display on battletv page [Link](https://rinish-api-testbed.stoplight.io/docs/codecharacter/b3A6MTQzMjUyMDE-get-user-matches)
- [x] Implement redux for battletv page

### CODE REVIEW

**Review by nishihere19 — APPROVED** [review]

**Comment by nishihere19** [comment]
Avatars are broken, apart from that everything looks fine to me.

**Comment by jaiakash** [comment]
- **Leaderboard** Data is fetched each time component is loaded.
- **BattleTV** Data is fetched only once when the components load for the first time.

> Avatars are broken, apart from that everything looks fine to me.

It would be integrated later when assets are decided.

**Review by PriyanshuDangi — CHANGES_REQUESTED** [review]
image

boxes are not same size

File: src/components/Leaderboard/Leaderboard.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  }, []);
+
+  useEffect(() => {
+    const endOffset = itemOffset + itemsPerPage;
+    setCurrentItems(items.slice(itemOffset, endOffset));
+    setPageCount(Math.ceil(items.length / itemsPerPage));
+  }, [itemOffset, itemsPerPage, items]);
+
+  const handlePageClick = (event: { selected: number }) => {
+    console.log(event);
```
Comment (PriyanshuDangi) [inline src/components/Leaderboard/Leaderboard.tsx:57]: remove

**Review by PriyanshuDangi — CHANGES_REQUESTED** [review]

File: src/components/BattleTV/BattleTV.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    name: string;
+  };
+}
+
+function PaginatedItems() {
+  const [pageCount, setPageCount] = useState(0);
+  const [itemOffset, setItemOffset] = useState(0);
+  const [currentItems, setCurrentItems] = useState([]);
+
+  const itemsPerPage = 4;
```
[the remark this answers is not in the document]
Comment (jaiakash) [reply src/components/BattleTV/BattleTV.tsx:32]: For testing purpose (pagination, just 4 so that it has multiple pages), i hardcoded it.

**Comment by jaiakash** [comment]
> Add loader till it loads the leaderboard also add a refresh button somewhere on leaderboard which will fetch the updated leaderboard from the backend

Done :+1:

File: src/components/BattleTV/BattleTV.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+  const { battletv, loading, hasErrors } = useAppSelector(battleTvSelector);
+
+  // initialize the redux hook
+  const dispatch = useAppDispatch();
+
+  if (battletv.length === 0) {
```
[the remark this answers is not in the document]
Comment (jaiakash) [reply src/components/BattleTV/BattleTV.tsx]: Added new field ```hasbeenFetched```.

**Review by PriyanshuDangi — APPROVED** [review]

**Pull request merged**